src/foraging_gui/Foraging.py

```diff
@@ -955,6 +957,27 @@ def _check_line_terminator(self, file_path):
                 return True
             else:
                 return False
+    
+    def _LoadSchedule(self):
+        if os.path.exists(self.Settings['schedule_path']):
+            schedule = pd.read_csv(self.Settings['schedule_path'])
+            box_list = []
+            for tower, box in itertools.product([1,2,3,6,7,8,9],['A','B','C','D']):
```

Comment:
Why do you constrain tower and box?

Reply:
Because there are rows in the schedule that are blank, have non-mouse information like dividers between 446 and 447 on the schedule. This just filters for rows that contain actual schedule information

Reply:
It also filters out boxes 4,5,6 which are for non-dynamic foraging tasks

Reply:
Can we just get all rows with a mouse ID? We will not load mouse IDs for non-dynamic foraging tasks. 

 Do you think we can load all the previous schedule information from somewhere? in this way, we can also use it in other rigs (e.g ephys rig).

Reply:
The problem is that `Mouse ID` also contains junk values from rows that are used for headers/dividers. Including entries like: "behcoverage6.17.png", "6/17/2024", "nan", "446", "447"

We can add boxes 4,5,6 if you want, but since those boxes never run the task it wont make any difference. 

Keeping a database of mice information is a nice idea, but lets make that a separate PR

Reply:
The first column contains the mouse IDs and other information, and we can easily extract rows with mouse IDs. Filtering on the mouse ID makes the code more generic, regardless of the device.

![image](https://github.com/AllenNeuralDynamics/dynamic-foraging-task/assets/109394934/a0c2780b-43eb-423a-ad2d-cdf4d4fdaf2d)

Reply:
> Keeping a database of mice information is a nice idea, but lets make that a separate PR

If it's very complex, it may not worth doing it recently.

Reply:
> The problem is that `Mouse ID` also contains junk values from rows that are used for headers/dividers. Including entries like: "behcoverage6.17.png", "6/17/2024", "nan", "446", "447"
> 
> We can add boxes 4,5,6 if you want, but since those boxes never run the task it wont make any difference.
> 
> Keeping a database of mice information is a nice idea, but lets make that a separate PR

There is no need to even fetch the rows with the mouse ID, we can just read all the rows and match the rows with the query mouse ID.

Reply:
@XX-Yin I've changed the section so now I just drop rows where either "Mouse ID" or "Box" are NaN.